theano/sandbox/gpuarray/__init__.py

```diff
@@ -64,8 +65,24 @@ def init_dev(dev, name=None):
     reg_context(name, context)
     pygpu_activated = True
     if config.print_active_device:
-        print("Mapped name %s to device %s: %s" % (name, dev, context.devname),
+        warn = None
+        cudnn_version = ""
+        if dev.startswith('cuda'):
+            try:
+                cudnn_version = dnn.version()
+                # 4100 should not print warning with cudnn 4 final.
+                if cudnn_version > 4100:
+                    warn = ("Your CuDNN version is more recent then Theano."
+                            " If you see problems, try updating Theano or"
+                            " downgrading CuDNN to version 4.")
+                cudnn_version = " (CuDNN version %s)" % cudnn_version
+            except Exception:
+                raise RuntimeError("CuDNN is mandatory with the GpuArray back-end.")
```

Comment:
The other option is to raise an error only when pooling or convolution is
used. If we don't, the user will get very slow execution. We can't expect
all user to see the warning and understand the consequence. Some software
generate too much output, a simple warning will get lost.

I don't like the idea to spent time making this selective error... We do
already enough stuff. Why do you want us to suport that case?


On Fri, Feb 5, 2016 at 1:23 PM, Pascal Lamblin <notifications@github.com>
wrote:

> In theano/sandbox/gpuarray/__init__.py
> <https://github.com/Theano/Theano/pull/4000#discussion_r52052728>:
>
> > @@ -64,8 +65,24 @@ def init_dev(dev, name=None):
> >      reg_context(name, context)
> >      pygpu_activated = True
> >      if config.print_active_device:
> > -        print("Mapped name %s to device %s: %s" % (name, dev, context.devname),
> > +        warn = None
> > +        cudnn_version = ""
> > +        if dev.startswith('cuda'):
> > +            try:
> > +                cudnn_version = dnn.version()
> > +                # 4100 should not print warning with cudnn 4 final.
> > +                if cudnn_version > 4100:
> > +                    warn = ("Your CuDNN version is more recent then Theano."
> > +                            " If you see problems, try updating Theano or"
> > +                            " downgrading CuDNN to version 4.")
> > +                cudnn_version = " (CuDNN version %s)" % cudnn_version
> > +            except Exception:
> > +                raise RuntimeError("CuDNN is mandatory with the GpuArray back-end.")
>
> I'm OK with not supporting convolutions without cudnn in the new back-end
> for a while, but elemwise, gemm, and so on should still be able to work.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/Theano/pull/4000/files#r52052728>.
>